### PR TITLE
[#4022] Throwing Generic Exceptions - LanguageGeneration

### PR DESCRIPTION
**Pull request by iscaltritti-sw**
Addresses # 4022

## Description

This PR replaces all 39 generic exceptions found within the libraries `Dialogs.Debugging`, `Dialogs.Declarative`, `LanguageGeneration`, `Configuration`, `Connector`, and `Parsers.LU` with proper exceptions according to the [Standard Exception Types](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/using-standard-exception-types).

## Specific Changes

- Replaces all 39 generic exceptions within:
  - libraries\Microsoft.Bot.Builder.Dialogs.Debugging\
  - libraries\Microsoft.Bot.Builder.Dialogs.Declarative\
  - libraries\Microsoft.Bot.Builder.LanguageGeneration\
  - libraries\Microsoft.Bot.Configuration\
  - libraries\Microsoft.Bot.Connector\
  - libraries\Parsers\Microsoft.Bot.Builder.Parsers.LU\
- Updates test `ConfigurationLoadAndSaveTests`, `TemplatesTest`, `TemplateDiagnosticTest`, and `LGGeneratorTests` to comply with the changes

## Testing
Here's a screenshot of the tests for these files working correctly with the changes applied:
image

### CODE REVIEW

**Review by santgr11 — APPROVED** [review]

**Review by sw-joelmut — APPROVED** [review]

File: libraries/Microsoft.Bot.Builder.Dialogs.Debugging/CodeModels/CodePoint.cs [hunk 1]
```diff
@@ -50,7 +50,7 @@ object ICodePoint.Evaluate(string expression)
             var result = exp.TryEvaluate(DialogContext.State);
             if (!string.IsNullOrEmpty(result.error))
             {
-                throw new Exception(result.error);
+                throw new ArgumentException(result.error);
```
Comment (ceciliaavila) [inline libraries/Microsoft.Bot.Builder.Dialogs.Debugging/CodeModels/CodePoint.cs:53]: Shouldn't this one be `InvalidOperationException`?  This is not an error with the argument itself.

**Review by ceciliaavila — CHANGES_REQUESTED** [review]

File: libraries/Microsoft.Bot.Builder.Dialogs.Debugging/CodeModels/CodePoint.cs (hunk 1 above)
Comment (iscaltritti-sw) [reply libraries/Microsoft.Bot.Builder.Dialogs.Debugging/CodeModels/CodePoint.cs:53]: The exception is the result of an evaluation of the argument `expression`, that's why I used `ArgumentException`.

Comment (ceciliaavila) [reply libraries/Microsoft.Bot.Builder.Dialogs.Debugging/CodeModels/CodePoint.cs:53]: But, in the rest of the PRs, you are using _InvalidOperationException_ for errors in expression evaluations.

Comment (iscaltritti-sw) [reply libraries/Microsoft.Bot.Builder.Dialogs.Debugging/CodeModels/CodePoint.cs]: Yes, there was a consistency issue. Thanks for pointing it out!

**Review by ceciliaavila — APPROVED** [review]

**Comment by ceciliaavila** [comment]
Promoted to MS in # 4954

**Pull request closed without merging**